proposals/build_tarball.md

```diff
@@ -0,0 +1,44 @@
+# Proposal: Build Image Tarball
+
+## Motivation
+
+Currently, Jib can build a container image to either a registry or a Docker daemon. Recently, there
+have been requests for the ability to build an image tarball directly to the filesystem so that the
+user may load it into docker via `docker load`, either manually or via a build system.
+
+## Goals
+
+* Build the image into a tarball
+* Minimize extra configuration
+
+## Constraints
+
+* The tarball will be saved to the project's build output directory with the name `jib-image.tar`
+
+## Intended Workflow
+
+The user will be able to output a `docker load`able image tarball by running `gradle jibBuildTar`
+for Gradle or `mvn jib:buildTar` for Maven. In the future, we may want to make the output path
+configurable.
+
+## Implementation
+
+The "build tarball" task is almost identical to the "build to docker daemon" task, except for the
+final step, which writes the final tarball to a file instead of piping it to a `docker load`
+command. To avoid duplicate code, a boolean parameter can be used in the steps runners to determine
+whether to build to a Docker daemon or build to a tarball.
+
+The following changes will be made to the code:
+1. Add a private output path configuration parameter to `BuildConfiguration`
+2. Rename:
+   - `BuildTarballAndLoadDockerStep` to `BuildTarballStep`
+   - `runBuildTarballAndLoadDockerStep` to `runBuildTarballStep`
+   - `waitOnBuildTarballAndLoadDockerStep` to `waitOnBuildTarballStep`
+3. Add a boolean parameter `doDockerLoad` to `runBuildTarballStep` and `BuildTarballStep`'s
```

Comment:
An alternative is to pass in a what-to-do-with-the-tarball-that-i-just-built parameter that defines the action to perform on the built tarball blob and have static factories on `BuildTarballStep` to use the right action.